src/components/molecules/AssetStatus.tsx

```diff
@@ -0,0 +1,51 @@
+import { DID } from '@oceanprotocol/lib'
+import axios, { CancelTokenSource } from 'axios'
+import React, { ReactElement, useEffect, useState } from 'react'
+import { useOcean } from '../../providers/Ocean'
+import { checkFile } from '../../utils/provider'
+import Status from '../atoms/Status'
+import Tooltip from '../atoms/Tooltip'
+import styles from './AssetStatus.module.css'
+
+export default function AssetStatus({
+  did
+}: {
+  did: DID | boolean
+}): ReactElement {
+  const { ocean } = useOcean()
+  const [isFileValid, setFileValid] = useState<boolean>()
+
+  useEffect(() => {
+    async function validateAsset(did: DID, source: CancelTokenSource) {
+      const ddo = await ocean.metadataCache.retrieveDDO(did)
```

Comment:
this won't work without a wallet connected. Please use the methods in `utils/aquarius.ts` which replicate these ocean.js methods, but directly hitting the Aquarius API

Reply:
I can change that but I think the provider file info call wouldn't work either without a wallet connected since we are using `config.providerUri` from `useOcean()` as well at `src/components/organisms/AssetActions/index.tsx`. This apply the same to https://github.com/oceanprotocol/market/pull/631

This mean to do the provider fileinfo check, we would need a wallet connected anyhow? 😞

Reply:
the `config` values are always present, without a wallet we load the default configured ones. But the `ocean` object is not and will only be initiated with a valid web3 provider, hence all methods under the `ocean` object are only available with a wallet. Which is why we replicate aquarius & provider methods for which a wallet is technically not required, like metadata query or the provider file check

Reply:
What about cases when a data asset use a different/custom provider? Can we get the custom providerUri from the DDO?

Reply:
yeah, good point, we actually need to if we want to cover different providers but I see you are already passing the respective `serviceEndpoint` further down, so that is the correct way of getting the provider url for each asset. Guess we need to strip out the path from the `serviceEndpoint` so we only get the protocol + host

Reply:
In that case, do this need to be updated to pass `serviceEndpoint`